oss_src/lambda/pylambda_worker.cpp

```diff
@@ -72,6 +72,8 @@ int EXPORT pylambda_worker_main(const std::string& root_path,
   LOG_DEBUG_WITH_PID("server_address = '" << server_address << "'");
   LOG_DEBUG_WITH_PID("parend pid = " << parent_pid);
 
+  size_t _last_line = __LINE__;
+#define __TRACK do { _last_line = __LINE__; } while(0)  
```

Comment:
Yes, you can; however, it messes up the ending ";", which is half the
reason to do it the C way.  It preserves macros as terminated statements
with required ";".

<http://www.dato.com>   Hoyt Koepke
Data Scientist
hoytak@dato.com   <https://twitter.com/datoinc>
<http://www.linkedin.com/company/graphlab>
<https://www.facebook.com/datoinc>
<http://www.youtube.com/channel/UC9T07WMdwB6IPf_dKXhVsCQ>


On Tue, Mar 15, 2016 at 1:08 PM, TobyRoseman <notifications@github.com>
wrote:

> In oss_src/lambda/pylambda_worker.cpp
> <https://github.com/dato-code/SFrame/pull/224#discussion_r56231514>:
>
> > @@ -72,6 +72,8 @@ int EXPORT pylambda_worker_main(const std::string& root_path,
> >    LOG_DEBUG_WITH_PID("server_address = '" << server_address << "'");
> >    LOG_DEBUG_WITH_PID("parend pid = " << parent_pid);
> >
> > +  size_t _last_line = __LINE__;
> > +#define __TRACK do { _last_line = __LINE__; } while(0)
>
> This is the C way of wrapping macros. From what I understand there is a
> more straight forward way to do this in C++. You can introduce arbitrary
> nesting where ever you like. So I believe you can just do:
>
> #define __TRACK do { _last_line = __LINE__; }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dato-code/SFrame/pull/224/files/15f77cd6727e9b2ab89a6616c3d2f2bcb4a076d0#r56231514>
>